fms/utils/serialization.py

```diff
@@ -29,7 +32,9 @@ def register_adapter(
     adapter: the class of the adapter. The class must accept one constructor
                 parameter, which will be a state dict (`OrderedDict`)
     """
-    sources: MutableMapping[str, Callable[[Mapping], Mapping]] = {}
+    sources: MutableMapping[
+        str, Callable[[Mapping, Optional[ModelConfig]], Mapping]
```

Comment:
if we made the ModelConfig kwargs-only would that eliminate the need for the more complicated type-hint? (e.g. `def foo(whatever, *, config=None)`